jax/_src/third_party/scipy/linalg.py

```diff
@@ -0,0 +1,127 @@
+import scipy.linalg
+
+from jax import jit, lax
+from jax._src.numpy import lax_numpy as jnp
+from jax._src.numpy.linalg import norm
+from jax._src.numpy.util import _wraps
+from jax._src.scipy.linalg import rsf2csf, schur
+
+_array_precision = {'i': 1, 'l': 1, 'f': 0, 'd': 1, 'F': 0, 'D': 1}
+
+eps = jnp.finfo(float).eps
+feps = jnp.finfo(jnp.single).eps
+
+
+def _maybe_real(A, B, tol=None):
+  """
+  Return either B or the real part of B, depending on properties of A and B.
+
+  The motivation is that B has been computed as a complicated function of A,
+  and B may be perturbed by negligible imaginary components.
+  If A is real and B is complex with small imaginary components,
+  then return a real copy of B.  The assumption in that case would be that
+  the imaginary components of B are numerical artifacts.
+
+  Parameters
+  ----------
+  A : ndarray
+      Input array whose type is to be checked as real vs. complex.
+  B : ndarray
+      Array to be returned, possibly without its imaginary part.
+  tol : float
+      Absolute tolerance.
+
+  Returns
+  -------
+  out : real or complex array
+      Either the input array B or only the real part of the input array B.
+
+  """
+  if jnp.isrealobj(A) and jnp.iscomplexobj(B):
+    if tol is None:
+      tol = {0: feps*1e3, 1: eps*1e6}[_array_precision[B.dtype.char]]
+    if jnp.allclose(B.imag, 0.0, atol=tol):
```

Comment:
This will not be compatible with jax transforms, so we should avoid it. We need to statically determine the output type. This will mean breaking with the scipy API, but that's necessary to make the API JAX-compatible in this case.

There are other examples of this, but I can't recall where they are right now.

Reply:
Thanks - I've removed the `_maybe_real` logic.